Plugin should support POD resource type for yml configuration file

### Description
According tp kubernets and openshift, there exists a resource type 'pod'. But fabric8 maven plugin does not recognize it when generating yml files using /src/main/fabric8/*-pod.yml. 

### Info
https://kubernetes.io/docs/user-guide/kubectl-overview/
https://docs.openshift.org/latest/dev_guide/templates.html

* f-m-p version : 3.4.1:
* Maven version (`mvn -v`) : 3.3.9

* Kubernetes / OpenShift setup and version : 
openshift v1.5.1+7b451fc
kubernetes v1.5.2+43a9be4

* If it's a bug, how to reproduce :
Example:  src/main/fabric8/node1-pod.yml

[INFO] F8: Using resource templates from C:\cygwin\home\someuser\dtmexamples\cloud\openshift\src\main\fabric8
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 7.328 s
[INFO] Finished at: 2017-06-14T14:13:36-04:00
[INFO] Final Memory: 38M/394M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal io.fabric8:fabric8-maven-plugin:3.4.1:resource (template) on project openshift: Execution template of goal io.fabric8:fabric8-maven-plugin:3.4.1:resource failed: Unknown type 'pod' for file node1-pod.yml. Must be one of : template, rs, bc, pr, role, rbr, ns, oauthclient, lr, pv, project, resourcequota, secret, istag, cronjob, sa, ds, crb, clusterrb, deploymentconfig, configmap, statefulset, pvc, deployment, limitrange, svc, rolebindingrestriction, cj, cm, is, daemonset, cr, rb, crole, rc, pb, clusterrole, route, service, namespace, rolebinding, rq, dc -> [Help 1]


* If it's a feature request, what is your use case :
Either fix to include -pod.yml or document to use some other resource type.

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -38,7 +38,7 @@ After this we will switch probably to real [Semantic Versioning 2.0.0](http://se
 * Fix 977: Allows user to define ConfigMap from XML and annotation
 * Fix 314: Do not execute GitEnricher if not Git repo
 * Fix 714: Merges the functionality of IANAServicePortNameEnricher and PortNameEnricher.
-
+* Fix 963: add support for Pod resource type in yaml config.
 ### 3.5.41 (2018-08-01)
 * Feature 1032: Improvements of the Vert.x Generator and enrichers
 * Fix 1313: Removed unused Maven goals. Please contact us if something's missing for you.

--- a/core/src/main/resources/META-INF/fabric8/kind-filename-type-mapping-default.adoc
+++ b/core/src/main/resources/META-INF/fabric8/kind-filename-type-mapping-default.adoc
@@ -108,4 +108,7 @@
 
 |Template
 |`template`
+
+|Pod
+|`pd`, `pod`
 |===

--- a/core/src/test/java/io/fabric8/maven/core/util/kubernetes/KindFilenameMapperUtilTest.java
+++ b/core/src/test/java/io/fabric8/maven/core/util/kubernetes/KindFilenameMapperUtilTest.java
@@ -53,6 +53,7 @@ public class KindFilenameMapperUtilTest {
         final Map<String, List<String>> expectedSerlializedContent = new HashMap<>();
         expectedSerlializedContent.put("ConfigMap", Arrays.asList("cm", "configmap"));
         expectedSerlializedContent.put("CronJob", Arrays.asList("cj", "cronjob"));
+        expectedSerlializedContent.put("Pod", Arrays.asList("pd", "pod"));
         assertThat(mappings).containsAllEntriesOf(expectedSerlializedContent);
 
     }

